### PR TITLE
Fixes to `calculate_mean_dirs`

### PR DESCRIPTION
**Pull request by tretomaszewski**
1. Using `torch.tensor(...).mean(dim=0)`has better precision than using python's built-in functions, especially for specialized datatypes such as "bfloat16".
2. 'mean_dir' key now outputs a tensor matching the model dimension rather than a single number.

**Warning**: Because of precision changes, this may cause different outputs than the original. Minimal tests have been conducted, but who knows with these things...

### CODE REVIEW

**Comment by tretomaszewski** [comment]
Now it is.
